### PR TITLE
Add facade infrastructure for remote relations

### PR DESCRIPTION
**Pull request by wallyworld**
The remote relations facade layers (api and apiserver) have methods added to support the underlying business logic defined in state. This is all boilerplate which will be used by the remote relations worker in a future PR.

QA: bootstrap lxd

### CODE REVIEW

File: api/remoterelations/remoterelations.go [hunk 1]
```diff
@@ -26,18 +26,146 @@ func NewClient(caller base.APICaller) *Client {
...
+// PublishLocalRelationChange publishes local relations changes to the
+// remote side offering those relations.
+func (c *Client) PublishLocalRelationChange(change params.RemoteRelationsChange) error {
+	args := params.RemoteRelationsChanges{
+		Changes: []params.RemoteRelationsChange{change},
+	}
+	var results params.ErrorResults
+	err := c.facade.FacadeCall("PublishLocalRelationChange", args, &results)
+	if err != nil {
+		return err
```
Comment (mjs) [inline api/remoterelations/remoterelations.go]: Trace (and similar below)

File: api/remoterelations/remoterelations.go [hunk 2]
```diff
@@ -26,18 +26,146 @@ func NewClient(caller base.APICaller) *Client {
...
+	err := c.facade.FacadeCall("ExportEntities", args, &results)
+	if err != nil {
+		return nil, err
+	}
+	if len(results.Results) != len(tags) {
+		return nil, errors.Errorf("expected %d result(s), got %d", len(tags), len(results.Results))
+	}
+	for _, r := range results.Results {
+		if r.Error != nil {
+			return nil, r.Error
```
Comment (mjs) [inline api/remoterelations/remoterelations.go]: This approach hides the source of the error (i.e. which entity) and also hides that some may have worked. Is that ok in this case?

Same for similar funcs below.

File: api/remoterelations/remoterelations_test.go [hunk 3]
```diff
@@ -93,8 +94,151 @@ func (s *remoteRelationsSuite) TestWatchLocalRelationUnits(c *gc.C) {
...
 		return nil
 	})
-	st := remoterelations.NewClient(apiCaller)
-	_, err := st.WatchLocalRelationUnits("relation-wordpress:db mysql:db")
+	client := remoterelations.NewClient(apiCaller)
+	_, err := client.WatchLocalRelationUnits("relation-wordpress:db mysql:db")
+	c.Check(err, gc.ErrorMatches, "FAIL")
+	c.Check(callCount, gc.Equals, 1)
+}
+
```
Comment (mjs) [inline api/remoterelations/remoterelations_test.go]: Tedious I know, but I don't see tests for the wrong number of return values, nor for the API call itself returning an error, for any of the new calls.

File: api/remoterelations/remoterelations.go (hunk 2 above)
Comment (wallyworld) [inline api/remoterelations/remoterelations.go]: Bah, I meant to change this. Thanks

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**